request.go

```diff
@@ -1,15 +1,23 @@
 package sarama
 
-type requestEncoder interface {
+import (
+	"fmt"
+)
+
+type requestBody interface {
 	encoder
+	decoder
 	key() int16
 	version() int16
 }
 
 type request struct {
+	length        int32
```

Comment:
length does not need to be a member of this struct; on encode it is implicit and on decode it has to be read in as a separate four-byte value anyways to know how much more to read